tests/morphology.cpp

```diff
@@ -180,12 +196,79 @@ BOOST_AUTO_TEST_CASE( h5_write_v1 )
     BOOST_CHECK( *types == *types2 );
 }
 
+BOOST_AUTO_TEST_CASE( h5_write_v11_glia )
+{
+    const std::string file( "glia.h5" );
+    boost::filesystem::remove( file );
+
+    boost::filesystem::path path( BBP_TESTDATA );
+    path /= "local/morphologies/01.07.08/h5/R-C010306G.h5";
+
+    const brion::Morphology morphology( path.string( ));
+    brion::floats perimeters;
+    brion::Vector4fsPtr points = morphology.readPoints();
+    perimeters.reserve( points->size( ));
+    for( size_t i = 0; i < points->size(); ++i )
+        perimeters.push_back( (*points)[i].w() * 4.f );
+
+    brion::Morphology glia( file, brion::FAMILY_GLIA );
+    glia.writePoints( *points );
+    glia.writeSections( *morphology.readSections( ));
+    glia.writeSectionTypes( *morphology.readSectionTypes( ));
+    glia.writePerimeters( perimeters );
+    BOOST_CHECK_THROW( glia.writeApicals( brion::Vector2is( )),
+                       std::runtime_error );
+
+    const brion::Morphology gliaRead( file );
+    BOOST_CHECK_EQUAL( gliaRead.getCellFamily(), brion::FAMILY_GLIA );
+    checkCloseArrays( *gliaRead.readPerimeters(), perimeters );
+
+    boost::filesystem::remove( file );
+}
+
+BOOST_AUTO_TEST_CASE( h5_write_invalid_glia )
+{
+    const std::string file( "glia.h5" );
+    boost::filesystem::remove( file );
+    brion::Morphology glia( file, brion::FAMILY_GLIA );
+    brion::floats perimeters;
+    BOOST_CHECK_THROW( glia.writePerimeters( perimeters ), std::runtime_error );
+    boost::filesystem::remove( file );
+}
+
+BOOST_AUTO_TEST_CASE( h5_write_v11_neuron )
+{
+    const std::string file( "neuron.h5" );
+    boost::filesystem::remove( file );
+
+    boost::filesystem::path path( BBP_TESTDATA );
+    path /= "local/morphologies/01.07.08/h5/R-C010306G.h5";
+
+    const brion::Morphology morphology( path.string( ));
+
+    const brion::floats perimeters;
+    brion::Morphology neuron( file, brion::FAMILY_NEURON );
+    neuron.writePoints( *morphology.readPoints( ));
+    neuron.writeSections( *morphology.readSections( ));
+    neuron.writeSectionTypes( *morphology.readSectionTypes( ));
+    neuron.writePerimeters( perimeters );
```

Comment:
The implementation shouldn't allow to write a file with n points and 0 perimeters. If you write points or perimeters and then you write the other dataset, they should be consistent. To leave it empty simply don't write it.